hack/jenkins/metadata-cache/metadata-cache.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Caches requests to the GCE metadata server.
+
+Reduces load on metadata server to once for most requests and once per
+~10m for access tokens.
+
+Installation:
+  sudo pip flask requests
+
+Usage:
+  sudo vim /etc/hosts  # Set metadata.google.internal to internal ip
+  sudo python metadata-cache.py  # sudo to listen on :80
+"""
+import collections
+import json
+import logging
+import logging.handlers
+import socket
+import threading
+import time
+
+import flask
+import requests
+
+app = flask.Flask(__name__)
+
+LOCK = threading.Lock()
+S = requests
```

Comment:
I am using S to stand for session, was flipping between requests.Session() (which preserves the session between requests) and requests (which opens a new session for each request)